packages/js/product-editor/src/components/label/label.tsx

```diff
@@ -0,0 +1,45 @@
+/**
+ * External dependencies
+ */
+import { createElement } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { Icon, help as helpIcon } from '@wordpress/icons';
+import { Tooltip } from '@wordpress/components';
+
+interface LabelProps {
+	label: string;
+	required?: boolean;
+	tooltip?: string;
+}
+
+export const Label: React.FC< LabelProps > = ( {
+	label,
+	required,
+	tooltip,
+} ) => {
+	return (
+		<div className="woocommerce-product-form-label__label">
+			{ label }
+			{ required && (
+				<span className="woocommerce-product-form-label__required">
+					{ /* translators: field 'required' indicator */ }
+					{ __( '*', 'woocommerce' ) }
+				</span>
+			) }
```

Comment:
This is text concatenation, which we should avoid, for localization. The original solution used `createInterpolateElement`, which while seeming a bit heavy-handed for this, does ensure that we are covered if a locale needed to switch the order of the required indicator and the label text.